### PR TITLE
Fixed LinkedIn URL + Created two new files 🐱💯 

### PR DESCRIPTION
**Pull request by boushrabettir**
- Created `opportunity.py` file which contains a class where one of the functions, `list_all_opportunities()` returns a `List[Opportunity]` as well as prints out all the jobs in the db.

- Created `utility.py` file which, has for now, has a `calculate_day_difference()` function. Will be used to place general reusable helper functions.

- Fixed LinkedIn web scraping issue (incorrect link as well as wrong dates being returned).

### CODE REVIEW

File: main.py
Comment (jjoeldaniel) [inline main.py:120]: Is it possible to narrow this down to avoid issues down the line where classes change?

Comment (boushrabettir) [reply main.py:120]: Sure!

**Comment by 26samaahmed** [comment]
gym bro made it!

**Review by 26samaahmed — APPROVED** [review]
Excellent organization skills.

File: utility.py
Comment (EthanThatOneKid) [inline utility.py]: Nit: `instantiates` implies a condition.
```suggestion
def instantiate_db_connection():
```

File: opportunity.py
Comment (EthanThatOneKid) [inline opportunity.py]: Return a list of the `Opportunity` dataclass.
```suggestion
def list_all_opportunities() -> List[Opportunity]:  # Return type of opportunity
```

File: opportunity.py
Comment (EthanThatOneKid) [inline opportunity.py]: Here is an example `Opportunity` dataclass.
```suggestion
class Opportunity:
    _company: str
    _title: str
    _location: str
    _link: str
    _processed: bool
```

File: opportunity.py
Comment (EthanThatOneKid) [inline opportunity.py:97]: Nit: Extra newlines.

File: opportunity.py
Comment (EthanThatOneKid) [inline opportunity.py]: Let's use the `Opportunity` dataclass and abstract the reading of rows into a list of opportunities. Here is a starting point.
```suggestion
def list_all_opportunities() -> List[Opportunity]:
    """Lists all opportunities in DB as well as returns them"""
    with utils.instantiates_db_connection() as connection:
        cursor = connection.cursor()
        cursor.execute(f"SELECT * FROM {table_name}")
        rows = cursor.fetchall()
        return read_opportunities(rows)

def list_filtered_opportunities() -> List[Opportunity]:
    """Returns a List[Opportunity] of job data that have a status of _processed = 0"""
    with utils.instantiates_db_connection() as connection:
        cursor = connection.cursor()
        cursor.execute(f"SELECT * FROM {table_name} WHERE _processed = 0 LIMIT 20")
        rows = cursor.fetchall()
        return read_opportunities(rows)

def read_opportunities(rows: List[tuple], debug: bool) -> List[Opportunity]:
    opps = []
    for row in rows: 
        _company, _title, _location, _link, _processed = row
        if debug:
            print("Company:", _company)
            print("Title:", _title)
            print("Location:", _location)
            print("Link:", _link)
            print("Processed:", _processed)
            print(" ")
        opp = Opportunity(_company, _title, _location, _link, _processed)
        opps.append(opp)
    return opps
```

**Review by EthanThatOneKid — COMMENTED** [review]
Thank you for continuing progress on this project! For now, I left a few comments about the `Opportunity` dataclass.

**Comment by boushrabettir** [comment]
Also, theres an [`enhancement issue`](https://github.com/acmcsufoss/acmOpportunities/issues/5#issue-1729312772) open that I'm going to reference since it's relevant.

**Comment by boushrabettir** [comment]
Theres no big "visual" difference, but with all my changes the message still was able to go through despite all my [changes](https://github.com/acmcsufoss/acmOpportunities/pull/7/commits/6bc7c18f27882352d99bcc58c69217548ce5edc8). This is just testing LinkedIn data. 

image

**Comment by boushrabettir** [comment]
Also to show the changes that GPT did, I forgot to send in this (Only used LinkedIn since either or with Indeed or LinkedIn it'd do the same thing):

image

File: .github/workflows/automate.yaml
Comment (EthanThatOneKid) [inline .github/workflows/automate.yaml]: Create a new issue as to not forget to follow up on this in the future.
```suggestion
    # TODO(#<ISSUE NUMBER>): Fix this time issue.
```

File: opportunity.py
Comment (EthanThatOneKid) [inline opportunity.py:146]: Does the introduction of `gpt_job_analyzer` resolve <https://github.com/acmcsufoss/acmOpportunities/issues/5>?

File: utility.py
Comment (EthanThatOneKid) [inline utility.py:1]: Does the introduction of `utility.py` resolve <https://github.com/acmcsufoss/acmOpportunities/issues/8>?

**Review by EthanThatOneKid — APPROVED** [review]
The progress on this PR is spectacular 🤯. I noticed it potentially resolves multiple issues, so be sure that all issues that are to be resolved by this PR are listed in the "Development" list. Be sure `main.py` runs smoothly as expected in manual local testing before merging.

File: opportunity.py
Comment (boushrabettir) [reply opportunity.py:146]: Yes, it does! I was waiting for the review before I closed it.

File: utility.py
Comment (boushrabettir) [reply utility.py:1]: I mean, it technically could be the case, but `utility.py` could be used for more general functions that could be helper functions versus having a db file reserved only for db helper functions but, we could implement those in `utility.py`.

**Comment by boushrabettir** [comment]
> . I noticed it potentially resolves multiple issues, so be sure that all issues that are to be resolved by this PR are listed in the "Development" list. Be sure `main.py` runs smoothly as expected in manual local testing before merging

There is... I have a photo here which I tested every function! (even ingesting new opportunities in the DB. I did not update their status though since they haven't been used by the ACM Bot.) 

image

File: .github/workflows/automate.yaml
Comment (boushrabettir) [reply .github/workflows/automate.yaml]: Resolved [here](https://github.com/acmcsufoss/acmOpportunities/pull/7/commits/e873aa82ae04037e5f55d8141c2e11e7f4be4012)

File: utility.py
Comment (EthanThatOneKid) [reply utility.py:1]: Would it help to move the database helpers into a `db.py`?

Comment (EthanThatOneKid) [reply utility.py:1]: If so, we can save that for a future PR and leave #8 open until we create a `db.py`.

**Pull request merged**
